src/code/z_demo.c

```diff
@@ -1125,264 +1159,291 @@ void func_800ECD7C(CutsceneContext* csCtx, u8** cutscenePtr, s16 index) {
  * - For each command list found, read the number of commands and loop over them, passing each one to the corresponding
  * function which handles the command, applying its effects and checking the frame range stored in the command.
  *
  * This function is invoked once per frame that a cutscene is active.
  *
- * TODO: consider changing the type of `cutscenePtr` to `uintptr_t` when this function matches.
+ * TODO: consider changing the type of `script` to `uintptr_t` when this function matches.
  */
-void Cutscene_ProcessCommands(PlayState* play, CutsceneContext* csCtx, u8* cutscenePtr) {
+void Cutscene_ProcessScript(PlayState* play, CutsceneContext* csCtx, u8* script) {
     s32 i;
     s32 j;
     s32 pad;
     s32 totalEntries;
     u32 cmdType;
     s32 cmdEntries;
     s32 pad2;
-    s32 cutsceneEndFrame;
-    CsCmdBase* cmd;
+    s32 csFrameCount;
+    void* cmd;
 
     // Read the command list count and the ending frame for this cutscene
-    bcopy(cutscenePtr, &totalEntries, sizeof(s32));
-    cutscenePtr += sizeof(s32);
-    bcopy(cutscenePtr, &cutsceneEndFrame, sizeof(s32));
-    cutscenePtr += sizeof(s32);
-
-    if (((u16)cutsceneEndFrame < csCtx->frames) && (play->transitionTrigger != TRANS_TRIGGER_START) &&
-        (csCtx->state != CS_STATE_4)) {
-        csCtx->state = CS_STATE_3;
+    bcopy(script, &totalEntries, sizeof(totalEntries));
+    script += sizeof(totalEntries);
+
+    bcopy(script, &csFrameCount, sizeof(csFrameCount));
+    script += sizeof(csFrameCount);
+
+    if ((csCtx->curFrame > (u16)csFrameCount) && (play->transitionTrigger != TRANS_TRIGGER_START) &&
+        (csCtx->state != CS_STATE_RUN_UNSTOPPABLE)) {
+        csCtx->state = CS_STATE_STOP;
         return;
     }
 
     // Loop over every command list
     for (i = 0; i < totalEntries; i++) {
         // Read the command type of the current command list.
-        bcopy(cutscenePtr, &cmdType, sizeof(u32));
-        cutscenePtr += sizeof(u32);
+        bcopy(script, &cmdType, sizeof(cmdType));
+        script += sizeof(cmdType);
 
-        // TODO: This should probably be added to the CutsceneCmd enum. Consider doing it when this function matches.
-        if (cmdType == 0xFFFFFFFF) {
+        if (cmdType == CS_CAM_STOP) {
             break;
         }
 
-        // Check special cases of command types. This are generic ActorActions
-        // Ranges: [0x64, 0x96), 0xC9, [0x1C2, 0x258)
-        if (((cmdType >= 100) && (cmdType < 150)) || (cmdType == 201) || ((cmdType >= 450) && (cmdType < 600))) {
-            for (j = 0; j < ARRAY_COUNT(D_801F4DC8); j = (s16)(j + 1)) {
-                if (D_801F4DC8[j] == (u16)cmdType) {
-                    func_800ECD7C(csCtx, &cutscenePtr, j);
-                    cmdType = -2;
+        // Check special cases of command types. This are generic ActorCues
+        if (((cmdType >= CS_CMD_ACTOR_CUE_100) && (cmdType <= CS_CMD_ACTOR_CUE_149)) ||
```

Comment:
this -> these

Reply:
is the comment needed anymore actually? I think you can just see from the condition that its a bunch of actor cue special cases

Reply:
No, I think it can go. It was useful when these were still hard-coded numbers, but the enum speaks for itself now